DatadogCore/Sources/Core/Context/DatadogContextProvider.swift

```diff
@@ -65,21 +62,12 @@ internal final class DatadogContextProvider {
         self.context = context
         self.receivers = []
         self.subscriptions = []
-        self.reader = KeyPathContextValueReader()
     }
 
     deinit {
         subscriptions.forEach { $0.cancel() }
     }
 
-    /// Reads current context.
-    ///
-    /// **Warning:** Must be called from the `queue`.
-    private func unsafeRead() -> DatadogContext {
-        reader.read(to: &self.context)
-        return context
-    }
```

Comment:
It was harder to make sense before given `context` is private property and we were passing the reference outside which can do whatever (essentially modify). Good clearance 🧹

Reply:
Although we are still leaking it in[ L100](https://github.com/DataDog/dd-sdk-ios/blob/maxep/remove-context-reader/DatadogCore/Sources/Core/Context/DatadogContextProvider.swift#L100) which isn't obvious for someone new jumping in.

Reply:
Yes, it wasn't properly documented..
Good catch on [L100](https://github.com/DataDog/dd-sdk-ios/blob/maxep/remove-context-reader/DatadogCore/Sources/Core/Context/DatadogContextProvider.swift#L100) 👍

Reply:
It's not actually a leak, it's the mutation block of the context. It would be great to make the context `Equitable`...

Reply:
that's something we must do, it is in my list
- a simple set to context is not a good indicator to trigger receivers. We must check the diff which eventually means  we need to have `Equatable` conformance.

Reply:
Agreed! The issue is with the `featureBaggages` and `userInfo` which is a free form dict.